spec/features/user_interacts_with_queue_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+feature "User interacts with the queue" do
+  scenario "user adds and reorders videos in the queue" do
+    comedies = Fabricate(:category)
+    south_park = Fabricate(:video, title: 'south park', category: comedies)
+    monk = Fabricate(:video, title: 'monk', category: comedies)
+    futurama = Fabricate(:video, title: 'futurama', category: comedies)
+
+    sign_in
+    
+    add_video_to_queue(monk)
+    expect_video_to_be_in_queue(monk)
+
+    visit video_path(monk)
+    expect_link_not_to_be_seen("+ My Queue")
+
+    add_video_to_queue(south_park)
+    add_video_to_queue(futurama)
+    
+    set_video_position(south_park, 1)
+    set_video_position(futurama, 2)
+    set_video_position(monk, 3)
+
+    update_queue
+
+    expect_video_position(south_park, 1)
+    expect_video_position(futurama, 2)
+    expect_video_position(monk, 3)
+  end
+
+  private
+
+  def expect_video_to_be_in_queue(video)
+    page.should have_content(video.title)
```

Comment:
I would not mix `should` syntax with `expect` - make sure you use just 1, while `expect` is latter and seemingly more preferred nowadays.